test/runtests.jl

```diff
@@ -16,13 +16,22 @@ if GROUP == "All" || GROUP == "Core"
         dst = zeros(Int, 4)
         bc = Broadcast.Broadcasted(+, (Broadcast.Broadcasted(*, (x, y)), x, y, x, y, x, y))
         bcref = copy(bc)
-        @test FastBroadcast.fast_materialize!(FastBroadcast.False(), dst, bc) == bcref
+        @test FastBroadcast.fast_materialize!(FastBroadcast.False(), FastBroadcast.True(), dst, bc) == bcref
+        @test FastBroadcast.fast_materialize!(FastBroadcast.False(), FastBroadcast.False(), dst, bc) == bcref
         dest = similar(bcref);
         @test (@.. dest = (x*y) + x + y + x + y + x + y) == bcref
         @test (@.. (x*y) + x + y + x + y + x + y) == bcref
         @test (@.. thread=true dest .+= (x*y) + x + y + x + y + x + y) ≈ 2bcref
         @test (@.. dest .-= (x*y) + x + y + x + y + x + y) ≈ bcref
         @test (@.. thread=true dest *= (x*y) + x + y + x + y + x + y) ≈ abs2.(bcref)
+
+        @test (@.. broadcast=false dest = (x*y) + x + y + x + y + x + y) == bcref
+        @test (@.. broadcast=false (x*y) + x + y + x + y + x + y) == bcref
+        @test (@.. broadcast=false thread=true dest .+= (x*y) + x + y + x + y + x + y) ≈ 2bcref
+        @test (@.. broadcast=false dest .-= (x*y) + x + y + x + y + x + y) ≈ bcref
+        @test (@.. broadcast=false thread=true dest *= (x*y) + x + y + x + y + x + y) ≈ abs2.(bcref)
```

Comment:
What if there's a broadcast when `broadcast=false`? Should we throw a readable error?

Reply:
Your call. What did the old `@..` do?

Reply:
It gave a really ugly error message :p so maybe we can do better. Could be a different PR though.